tests/testthat/test-make.R

```diff
@@ -98,3 +108,273 @@ test_that("graph_from_literal(simplify = FALSE)", {
+  expect_equal(ecount(graph_unnamed_args), 1)
+})
+
+test_that("make_empty_graph gives an error for invalid arguments", {
+  expect_snapshot(make_empty_graph(NULL), error = TRUE)
+  expect_warning(expect_error(make_empty_graph("spam")))
+})
+
+test_that("make_graph_atlas works", {
+  atlas_124 <- graph_from_atlas(124)
+  expect_isomorphic(atlas_124, make_graph(c(1, 2, 2, 3, 3, 4, 4, 5, 1, 5, 1, 3, 2, 6),
+    directed = FALSE
+  ))
+  atlas_234 <- graph_from_atlas(234)
+  expect_isomorphic(atlas_234, make_graph(c(1, 6, 2, 6, 3, 6, 4, 6, 5, 6),
+    n = 7,
+    directed = FALSE
+  ))
+})
+
+test_that("make_chordal_ring works", {
+  chord <- make_chordal_ring(
+    15,
+    matrix(c(3, 12, 4, 7, 8, 11), nr = 2)
+  )
+  expect_equal(degree(chord), rep(6, 15))
+})
+
+test_that("make_line_graph works", {
+  graph_ring_n5 <- make_ring(n = 5)
+  graph_line_n5 <- make_line_graph(graph_ring_n5)
+  expect_isomorphic(graph_line_n5, graph_ring_n5)
+})
+
+test_that("make_de_bruijn_graph works", {
+  de_bruijn21 <- make_de_bruijn_graph(2, 1)
+  de_bruijn22 <- make_de_bruijn_graph(2, 2)
+  de_bruijn21_line <- make_line_graph(de_bruijn21)
+
+  expect_isomorphic(de_bruijn21_line, make_graph(c(
+    1, 1, 3, 1, 1, 2, 3, 2, 2, 3,
+    4, 3, 2, 4, 4, 4
+  )))
+  expect_isomorphic(de_bruijn22, de_bruijn21_line)
+})
+
+test_that("make_bipartite_graph works", {
+  inc_mat_rand <- matrix(sample(0:1, 35, replace = TRUE, prob = c(3, 1)), ncol = 5)
+  bip_from_inc <- graph_from_biadjacency_matrix(inc_mat_rand)
+
+  edges <- unlist(sapply(seq_len(nrow(inc_mat_rand)), function(x) {
+    w <- which(inc_mat_rand[x, ] != 0) + nrow(inc_mat_rand)
+    if (length(w) != 0) {
+      as.vector(rbind(x, w))
+    } else {
+      numeric()
+    }
+  }))
+  bip_from_make <- make_bipartite_graph(seq_len(nrow(inc_mat_rand) + ncol(inc_mat_rand)) > nrow(inc_mat_rand), edges)
+  inc_mat_bip <- as_biadjacency_matrix(bip_from_make)
+
+  expect_equal(inc_mat_bip, inc_mat_rand, ignore_attr = TRUE)
+})
+
+test_that("make_bipartite_graph works with vertex names", {
+  types <- c(0, 1, 0, 1, 0, 1)
+  names(types) <- LETTERS[1:length(types)]
+  edges <- c("A", "B", "C", "D", "E", "F", "A", "D", "D", "E", "B", "C", "C", "F")
+  bip_grap <- make_bipartite_graph(types, edges)
+
+  expect_equal(V(bip_grap)$name, c("A", "B", "C", "D", "E", "F"), ignore_attr = TRUE)
+  expect_equal(V(bip_grap)$type, c(FALSE, TRUE, FALSE, TRUE, FALSE, TRUE), ignore_attr = TRUE)
+
+  expect_error(make_bipartite_graph(types, c(edges, "Q")), "edge vector contains a vertex name that is not found")
+})
+
+test_that("make_full_bipartite_graph works", {
+  full_bip_star <- make_full_bipartite_graph(5, 1)
+  expect_isomorphic(full_bip_star, make_star(6, "undirected"))
+
+  full_bip <- make_full_bipartite_graph(5, 5)
+  expect_equal(vcount(full_bip), 10)
+  expect_equal(ecount(full_bip), 25)
+})
+
+test_that("make_kautz_graph works", {
+  kautz <- make_kautz_graph(2, 3)
+  expect_equal(kautz$name, "Kautz graph 2-3")
```

Comment:
use testthat::expect_named?

Reply:
doesnt work here properly because igraph objects are lists but not really named lists